auth/service_test.go

```diff
@@ -1975,7 +1975,7 @@ func TestChangeDomainStatus(t *testing.T) {
 				Status: &disabledStatus,
 			},
 			retreieveByIDErr: repoerr.ErrNotFound,
-			err:              svcerr.ErrAuthorization,
+			err:              svcerr.ErrNotFound,
```

Comment:
@WashingtonKK, Why we are changing this error type?

Reply:
If the domain is empty or not avialable with MG, then it should return Authorization Error, 
We should not return 404 , So I'm suggesting following changes in two functions 


1.) 

https://github.com/washingtonkk/magistrala/blob/b50c99851ec8b3bbb23b26476896c1aad8f12fde/auth/service.go#L187-L205
Suggestion  for the above function 
```go
func (svc service) checkPolicy(ctx context.Context, pr PolicyReq) error {
	// Domain status is required for if user sent authorization request on things, channels, groups and domains
	if pr.SubjectType == UserType && (pr.ObjectType == GroupType || pr.ObjectType == ThingType || pr.ObjectType == DomainType) {
		if err := svc.checkDomain(ctx, pr.SubjectType, pr.Subject, pr.Domain); err != nil {
			return err
		}
	}
	if err := svc.agent.CheckPolicy(ctx, pr); err != nil {
		return errors.Wrap(svcerr.ErrAuthorization, err)
	}
	return nil
}
```

2.) 
https://github.com/washingtonkk/magistrala/blob/b50c99851ec8b3bbb23b26476896c1aad8f12fde/auth/service.go#L207-L240
Suggestion for above function 

```
func (svc service) checkDomain(ctx context.Context, subjectType, subject, domainID string) error {
	if err := svc.agent.CheckPolicy(ctx, PolicyReq{
		Subject:     subject,
		SubjectType: subjectType,
		Permission:  MembershipPermission,
		Object:      domainID,
		ObjectType:  DomainType,
	}); err != nil {
		return svcerr.ErrDomainAuthorization
	}

	d, err := svc.domains.RetrieveByID(ctx, domainID)
	if err != nil {
		return errors.Wrap(errors.ErrUnidentified, err)
	}

	switch d.Status {
	case EnabledStatus:
	case DisabledStatus:
		if err := svc.agent.CheckPolicy(ctx, PolicyReq{
			Subject:     subject,
			SubjectType: subjectType,
			Permission:  AdminPermission,
			Object:      domainID,
			ObjectType:  DomainType,
		}); err != nil {
			return svcerr.ErrDomainAuthorization
		}
	case FreezeStatus:
		if err := svc.agent.CheckPolicy(ctx, PolicyReq{
			Subject:     subject,
			SubjectType: subjectType,
			Permission:  AdminPermission,
			Object:      MagistralaObject,
			ObjectType:  PlatformType,
		}); err != nil {
			return svcerr.ErrDomainAuthorization
		}
	default:
		return svcerr.ErrDomainAuthorization
	}

	return nil
}
```

Reply:
This makes sense @arvindh123, it is a great idea. Let me implement it

Reply:
Thank you @WashingtonKK